lib/Widgets/Utils.vala

```diff
@@ -204,6 +204,68 @@ public static string markup_accel_tooltip (string[]? accels, string? description
     return string.joinv ("\n", parts);
 }
 
+private double contrast_ratio (Gdk.RGBA bg_color, Gdk.RGBA fg_color) {
+    var bg_luminance = get_luminance (bg_color);
+    var fg_luminance = get_luminance (fg_color);
+
+    if (bg_luminance > fg_luminance) {
+        return (bg_luminance + 0.05) / (fg_luminance + 0.05);
+    }
+
+    return (fg_luminance + 0.05) / (bg_luminance + 0.05);
+}
+
+private double get_luminance (Gdk.RGBA color) {
+    var red = sanitize_color (color.red) * 0.2126;
+    var green = sanitize_color (color.green) * 0.7152;
+    var blue = sanitize_color (color.blue) * 0.0722;
+
+    return (red + green + blue);
+}
+
+private double sanitize_color (double color) {
+    if (color <= 0.03928) {
```

Comment:
Should probably comment where this magic number is coming from and/or use a const?

Reply:
@danrabbit this came from https://github.com/danrabbit/harvey/blob/57dcb07e55b7e33eb0947de720b95117228bab35/src/MainWindow.vala#L268 originally. Do you know how to explain these numbers? :sweat_smile:

Reply:
Not Daniel, but I can say that they're probably not "magic" as in what works or doesn't but are part of an algorithm that dictates those as a required part. You probably don't need to have constants for all these at all.
For example `get_luminance` uses a formula directly from a wiki page: https://en.wikipedia.org/wiki/Relative_luminance which in turn uses `sanitize_color` from https://en.wikipedia.org/wiki/SRGB#The_reverse_transformation.

Reply:
@donadigo my problem is that they're not super obvious to passers-by, so I wonder if documenting them better here (even just with a comment) would be worthwhile.

Reply:
Yeah these numbers come from:
https://www.w3.org/TR/WCAG20/#contrast-ratiodef
https://www.w3.org/TR/WCAG20/#relativeluminancedef

I think it would be worth commenting with links to either WCAG or Wikipedia